examples/assemblyscript/lib/warduino.ts

```diff
@@ -0,0 +1,105 @@
+@external("env", "wifi_status")         export declare function wifi_status(): i32;
+@external("env", "wifi_localip")        declare function _wifi_localip(buff: ArrayBuffer, buffer_size: u32): i32;
+
+@external("env", "http_get")            declare function _http_get(url: ArrayBuffer, url_len: u32, buffer: ArrayBuffer, buffer_size: u32): i32;
+@external("env", "http_post")           declare function _http_post(url: ArrayBuffer, url_len: u32,
+                                                                    body: ArrayBuffer, body_len: u32,
+                                                                    content_type: ArrayBuffer, content_type_len: u32,
+                                                                    authorization: ArrayBuffer, authorization_len: u32,
+                                                                    buffer: ArrayBuffer, buffer_size: u32): i32;
+
+@external("env", "mqtt_init")        declare function  _mqtt_init(server: ArrayBuffer, length: i32, port: u32): void;
+@external("env", "mqtt_connect")     declare function  _mqtt_connect(client_id: ArrayBuffer, length: i32): i32;
+@external("env", "mqtt_connected")   declare function  _mqtt_connected(): i32;
+@external("env", "mqtt_state")       export declare function  mqtt_state(): i32;
+@external("env", "mqtt_publish")     declare function  _mqtt_publish(topic: ArrayBuffer, length: i32, payload: ArrayBuffer, length: i32): i32;
+@external("env", "mqtt_subscribe")   declare function  _mqtt_subscribe(topic: ArrayBuffer, length: i32, fn: (topic: ArrayBuffer, topic_length: i32, payload: ArrayBuffer, payload_length: i32) => void): i32;
+@external("env", "mqtt_unsubscribe") declare function  _mqtt_unsubscribe(topic: ArrayBuffer, length: i32, fn: (topic: ArrayBuffer, topic_length: i32, payload: Arraybuffer, payload_length: i32) => void): i32;
+@external("env", "mqtt_loop")        export declare function  mqtt_loop(): i32;
+
+export const LOW: u32 = 0;
+export const HIGH: u32 = 1;
+
+export const CHANGED: u32 = 1;
+export const FALLING: u32 = 2;
+export const RISING: u32 = 3;
+
+export const INPUT: u32 = 0x0;
+export const OUTPUT: u32 = 0x2;
+
+export const WL_CONNECTED: u32 = 3;
+
+export function sleep(s: u32): void {
+    delay(s * 1000);
+}
+
+export function interrupt_on(pin: u32, mode: u32, fn: () => void): void {
+    subscribe_interrupt(pin, fn, mode);
+}
+
+export function print(text: string): void {
+    _print(String.UTF8.encode(text, true), String.UTF8.byteLength(text, true));
+}
+
+export function wifi_connect(ssid: string, password: string): void {
+    _wifi_connect(String.UTF8.encode(ssid, true), String.UTF8.byteLength(ssid, true),
+                  String.UTF8.encode(password, true), String.UTF8.byteLength(password, true));
+}
+
+export function wifi_localip(): string {
+    let localip = new ArrayBuffer(20);
```

Comment:
```suggestion
    let localip = new ArrayBuffer(46);
```

Maybe this should be set tot 46 to also support ipv6 adresses (https://stackoverflow.com/a/166157)

Reply:
Added in 1fe5979